Sources/NIO/EventLoop.swift

```diff
@@ -92,21 +94,45 @@ public final class RepeatedTask {
     ///  This means immediate cancellation is not guaranteed.
     ///
     /// The safest way to cancel is by using the passed reference of `RepeatedTask` inside the task closure.
-    public func cancel() {
+    ///
+    /// If the promise parameter is not `nil`, the passed promise is fulfilled when cancellation is complete.
+    /// Passing a promise does not prevent fulfillment of any promise provided on original task creation.
+    public func cancel(andNotify promise: EventLoopPromise<Void>? = nil) {
```

Comment:
Would you mind naming this parameter just `promise`? That would look more consistent with our other APIs.
Also: adding `- parameters` to the API docs here would be great

Reply:
Done!